tests/components/reolink/test_switch.py

```diff
@@ -0,0 +1,81 @@
+"""Test the Reolink switch platform."""
+
+from unittest.mock import MagicMock, patch
+
+from homeassistant.components.reolink import const
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er, issue_registry as ir
+
+from .conftest import TEST_UID
+
+from tests.common import MockConfigEntry
+
+
+async def test_cleanup_hdr_switch_(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    reolink_connect: MagicMock,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test cleanup of the HDR switch entity."""
+    original_id = f"{TEST_UID}_hdr"
+    domain = Platform.SWITCH
+
+    reolink_connect.channels = [0]
+    reolink_connect.supported.return_value = True
+
+    entity_registry.async_get_or_create(
+        domain=domain,
+        platform=const.DOMAIN,
+        unique_id=original_id,
+        config_entry=config_entry,
+        suggested_object_id=original_id,
+        disabled_by=er.RegistryEntryDisabler.USER,
+    )
+
+    assert entity_registry.async_get_entity_id(domain, const.DOMAIN, original_id)
```

Comment:
I have a personal ick for the `const.DOMAIN`, but that's personal preference

Reply:
I also find it very confusing that in entity_registry.async_get_or_create( the domain is supposed to be "switch" and the platform supposed to be "reolink".

In my head this is the wrong since I always though the domain is Reolink and platforms are switch, select etc....
But anyway.

Lets leave this for some other PR.

Reply:
It also confused me many times lol